zod/src/zodv4.ts

```diff
@@ -0,0 +1,134 @@
+import { toNestErrors, validateFieldsNatively } from '@hookform/resolvers';
+import {
+  FieldError,
+  FieldErrors,
+  FieldValues,
+  Resolver,
+  ResolverError,
+  ResolverSuccess,
+  appendErrors,
+} from 'react-hook-form';
+import { ZodError } from 'zod-v4/v4';
+import * as core from 'zod-v4/v4/core';
+
+function parseErrorSchema(
+  zodErrors: core.$ZodIssue[],
+  validateAllFieldCriteria: boolean,
+) {
+  const errors: Record<string, FieldError> = {};
+  for (; zodErrors.length; ) {
+    const error = zodErrors[0];
+    const { code, message, path } = error;
+    const _path = path.join('.');
+
+    if (!errors[_path]) {
+      if (error.code === 'invalid_union') {
+        const unionError = error.errors[0][0];
+
+        errors[_path] = {
+          message: unionError.message,
+          type: unionError.code,
+        };
+      } else {
+        errors[_path] = { message, type: code };
+      }
+    }
+
+    if (error.code === 'invalid_union') {
+      error.errors.forEach((unionError: any[]) =>
+        unionError.forEach((e) =>
+          zodErrors.push({
+            ...e,
+            path: [...path, ...e.path],
+          }),
+        ),
+      );
+    }
+
+    if (validateAllFieldCriteria) {
+      const types = errors[_path].types;
+      const messages = types && types[error.code];
+
+      errors[_path] = appendErrors(
+        _path,
+        validateAllFieldCriteria,
+        errors,
+        code,
+        messages
+          ? ([] as string[]).concat(messages as string[], error.message)
+          : error.message,
+      ) as FieldError;
+    }
+
+    zodErrors.shift();
+  }
+
+  return errors;
+}
+
+export function zodResolver<Input extends FieldValues, Context, Output>(
+  schema: core.$ZodType<Output, Input>,
+  schemaOptions?: Partial<core.ParseContext<core.$ZodIssue>>,
+  resolverOptions?: {
+    mode?: 'async' | 'sync';
+    raw?: false;
+  },
+): Resolver<Input, Context, Output>;
```

Comment:
it seems like the overload for `raw: true` was incorrectly removed, which breaks the types when this option is passed @alexcraviotto

Reply:
see https://github.com/react-hook-form/resolvers/pull/776#discussion_r2122258830